pages/guides/developing/encrypted-vod.en-US.mdx

```diff
@@ -0,0 +1,186 @@
+if err != nil {
+  log.Fatalf("Error writing encrypted data to output file: %v", err)
+}
+```
+<Callout>
+  Currently Livepeer supports only video content encrypted using AES-CBC with PKCS#7 padding.
+</Callout>
+
+## Step 3: Retrieve the Livepeer Public Key
+
+After obtaining an encryption key and encrypting a video file with it, the next step is to retrieve the Livepeer Public Key. This key will be used to encrypt our encryption key, which can then be shared with Livepeer when creating a video asset.
+
+```go
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"log"
+)
+
+type Response struct {
+	PublicKey string `json:"public_key"`
+}
+
+func main() {
+	publicKey, err := fetchPublicKey()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Println(publicKey)
+}
+
+func fetchPublicKey() (string, error) {
+	// Create a new request using http
+	req, err := http.NewRequest("GET", "https://livepeer.studio/api/access-control/public-key", nil)
+	if err != nil {
+		return "", err
+	}
+
+	// Add authorization header to the req
+	req.Header.Add("Authorization", "Bearer XXXX-XXXXXXX-XXXXXX-XXXXX")
+
+	// Send req using http Client
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	
+	defer resp.Body.Close()
+
+	// Check the server response for errors
+	if resp.StatusCode != 200 {
+		return "", fmt.Errorf("server returned: %s", resp.Status)
+	}
+
+	// Read the response body
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	// Unmarshal the JSON response
+	var response Response
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		return "", err
+	}
+
+	return response.PublicKey, nil
+}
+```
+
+## Step 4: Encrypt the encryption key
+
+Now, we can use asymmetric encryption with the Livepeer Public Key to encrypt our Encryption Key. The resulting data is then encoded in base64 before being sent to Livepeer along with the video file.
+
+```go filename="main.go"
+// Encrypt the key with the RSA public key
+encryptedKey, err := rsa.EncryptOAEP(sha256.New(), rand.Reader, publicKey, encryptionKey, nil)
+if err != nil {
+  log.Fatalf("Error encrypting key: %v", err)
+}
+
+// Save encrypted key as a base64 encoded string
+encodedKey := base64.StdEncoding.EncodeToString(encryptedKey)
+```
+
+## Step 5: Upload the video file or Import it from decentralized storages
+
+We're all set to upload the video file. There are two ways to do this:
+
+- Request a direct upload and add the `encryptedKey` to the payload under the `encryption` object.
+- Import it from a decentralized or external storage and add the `encryptedKey` to the payload under the `encryption` object.
+
+When calling the `import` API:
+```json
+{
+    "url":"ipfs://QmaxiZYQhdDQrHrS3rnPHfvdJx8XfuVLv2nGv4y8cdVUnM",
+    "name":"import_from_ipfs",
+    "encryption":{
+        "encryptedKey":"<YOUR_BASE_64_ENCODED_ENCRYPTED_ENCRYPTION_KEY>"
+    }
+}
+```
+
+When calling the `request-upload` API:
```

Comment:
We can them "upload via URL" and "direct upload" today. though these should probably be API call examples (js and/or bash)